### PR TITLE
Containerize `TestDaemonNoSpaceLeftOnDeviceError`

### PR DESCRIPTION
**Pull request by cpuguy83**
Why? Because the `mount` here will sometimes fail when run in
`debian:jessie`, which is what the environrment hosting the test suite
is running if run from the `Makefile`.
Also, why the heck not containerize it, all the things.

This also makes this test succeed on docker4mac.

### CODE REVIEW

**Comment by runcom** [comment]
It fails when containerized in Fedora also :(

**Comment by cpuguy83** [comment]
@runcom This fails or the test before this change fails?

**Comment by cpuguy83** [comment]
Adding this to 1.11.1 since we need this for CI to fully pass.

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -1909,27 +1910,24 @@ func (s *DockerDaemonSuite) TestBridgeIPIsExcludedFromAllocatorPool(c *check.C)
...
 
-	cmd = exec.Command("mount", "-t", "ext4", "-no", "loop,rw", "/tmp/testfs.img", "/tmp/testfs-mount")
-	out, err = cmd.CombinedOutput()
-	c.Assert(err, checker.IsNil, check.Commentf(string(out)))
+	// create a 2MiB image and mount it as graph root
+	// Why in a container? Because `mount` sometimes behaves weirdly and often fails outright on this test in debian:jessie (which is what the test suite runs under if run from the Makefile)
+	dockerCmd(c, "run", "--rm", "-v", testDir+":/test", "busybox", "sh", "-c", "dd of=/test/testfs.img bs=1M seek=2 count=0")
+	out, _, err := runCommandWithOutput(exec.Command("mkfs.ext4", "-F", filepath.Join(testDir, "testfs.img"))) // `mkfs.ext4` is not in busybox
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	dockerCmd(c, "run", "--privileged", "--rm", "-v", testDir+":/test:shared", "busybox", "sh", "-c", "mkdir -p /test/test-mount && mount -t ext4 -no loop,rw /test/testfs.img /test/test-mount")
```
Comment (MHBauer) [inline integration-cli/docker_cli_daemon_test.go]: Let me check my understanding. The trick here is that the mount is shared, and the container is running privileged, so the mount inside the container appears outside the container for later use. Correct?

Comment (cpuguy83) [inline integration-cli/docker_cli_daemon_test.go]: Yes.

**Comment by MHBauer** [comment]
Amusing and completely awesome. Why not containerize all the things?

LGTM

**Comment by LK4D4** [comment]
LGTM

**Comment by tonistiigi** [comment]
The test passed on overlay daemon here https://jenkins.dockerproject.org/job/Tonis%20Docker%20debugginig/label=overlay/14/console

LGTM

**Pull request merged**